src/server/address_space_internal.cpp

```diff
@@ -295,6 +295,64 @@ namespace OpcUa
       return StatusCode::BadAttributeIdInvalid;
     }
 
+    void AddressSpaceInMemory::SetMethod(const NodeId& node, std::function<std::vector<OpcUa::Variant> (std::vector<OpcUa::Variant> arguments)> callback)
+    {
+      NodesMap::iterator it = Nodes.find(node);
```

Comment:
I am in doubts and because of new responsibility of address sspace. I'd
like prefer separate class for call service. ☺
24 апр. 2015 г. 11:17 пользователь "ORD" <notifications@github.com> написал:

> In src/server/address_space_internal.cpp
> <https://github.com/FreeOpcUa/freeopcua/pull/132#discussion_r29028785>:
>
> > @@ -295,6 +295,64 @@ namespace OpcUa
> >        return StatusCode::BadAttributeIdInvalid;
> >      }
> >
> > +    void AddressSpaceInMemory::SetMethod(const NodeId& node, std::function<std::vector<OpcUa::Variant> (std::vector<OpcUa::Variant> arguments)> callback)
> > +    {
> > +      NodesMap::iterator it = Nodes.find(node);
>
> thanks! Do not merge that. If you agree with the current code I will add
> at least one more commit to allow to use that implementation from server
> side. Otherwise it is a bit useless :-)
> … <#14cea4995145e07f_>
> On Thu, 23 Apr 2015 at 21:23 Alexander Rykovanov <notifications@github.com>
> wrote: In src/server/address_space_internal.cpp <
> https://github.com/FreeOpcUa/freeopcua/pull/132#discussion_r28994646>: >
> @@ -295,6 +295,64 @@ namespace OpcUa > return
> StatusCode::BadAttributeIdInvalid; > } > > + void
> AddressSpaceInMemory::SetMethod(const NodeId& node,
> std::function<std::vector<OpcUa::Variant> (std::vector<OpcUa::Variant>
> arguments)> callback) > + { > + NodesMap::iterator it = Nodes.find(node);
> This is not thread safe. — Reply to this email directly or view it on
> GitHub <https://github.com/FreeOpcUa/freeopcua/pull/132/files#r28994646>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FreeOpcUa/freeopcua/pull/132/files#r29028785>.
>